core/src/main/scala/org/apache/spark/memory/MemoryPool.scala

```diff
@@ -36,14 +36,14 @@ private[memory] abstract class MemoryPool(lock: Object) {
    * Returns the current size of the pool, in bytes.
    */
   final def poolSize: Long = lock.synchronized {
-    _poolSize
+    return _poolSize
```

Comment:
This is very strange. Is it because the variable is annotated with "GuardedBy"

Reply:
No, removing GuardedBy made no difference. As far as I can tell this is due to immediate return value inside a synchronized block -- removing that gets rid of the boxing/unboxing of course. Looks like a scala compiler bug that should be reported upstream. Btw the decompiler shows this for original code:
```java
public final long poolSize()
  {
    synchronized (this.lock)
    {
      Long localLong = BoxesRunTime.boxToLong(this._poolSize);return BoxesRunTime.unboxToLong(localLong);
    }
  }
```

Reply:
Seems like this was fixed here: https://github.com/scala/scala/commit/92d1af11b04b4f7c8aafd4ff911bf747eb1029aa but not in 2.11.x

The issue here https://github.com/scala/bug/issues/5326 indicates that "scalac -optimize" avoids the boxing. However, the "-optimize" takes quite long to compile so was removed. We could enable that for production release build that should also be used for hydra testing. What do you say?